src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ArrayTrailingCommaCheck.java

```diff
@@ -29,9 +29,21 @@
  * </p>
  * <p>
  * Rationale: Putting this comma in make is easier to change the
- * order of the elements or add new elements on the end.
+ * order of the elements or add new elements on the end. Main benefit of a trailing
+ * comma is that when you add new entry to an array, no surrounding lines are changed.
  * </p>
  * <p>
+ * The check allows leaving out the comma at the end if either left
+ * or right curly braces are on the same line as the last element of the array.
```

Comment:
Yes. This case is like "Trailing comma not needed, line needs to be modified anyway" , there should be no violation.
This is just another example that it is hard to predict behavior of Check now, so xdoc should be extended